tests/php/modules/shortcodes/test-class.dailymotion.php

```diff
@@ -141,4 +148,75 @@ public function test_shortcodes_dailymotion_params() {
 		$this->assertContains( 'ui-theme=dark', $shortcode_content );
 	}
 
+	/**
+	 * Gets the test data for test_shortcodes_dailymotion_amp().
+	 *
+	 * @return array The test data.
+	 */
+	public function get_dailymotion_amp_data() {
+		$id             = 26423151;
+		$default_height = 471;
+
+		return array(
+			'no_attribute'            => array(
+				'[dailymotion]',
+				'<!--Dailymotion error: bad or missing ID-->',
+			),
+			'plain_id'                => array(
+				'[dailymotion ' . $id . ']',
+				'<amp-dailymotion data-videoid="'. $id .'" layout="responsive" width="' . self::CONTENT_WIDTH . '" height="' . $default_height .'"></amp-dailymotion>',
+			),
+			'id_value_as_attribute'   => array(
+				'[dailymotion id=' . $id . ']',
+				'<amp-dailymotion data-videoid="'. $id .'" layout="responsive" width="' . self::CONTENT_WIDTH . '" height="' . $default_height .'"></amp-dailymotion>',
+			),
+			'dailymotion_value'       => array(
+				'[dailymotion=' . $id . ']',
+				'<amp-dailymotion data-videoid="'. $id .'" layout="responsive" width="' . self::CONTENT_WIDTH . '" height="' . $default_height .'"></amp-dailymotion>',
+			),
+			'width_in_attributes'     => array(
+				'[dailymotion ' . $id . ' width=300]',
+				'<amp-dailymotion data-videoid="'. $id .'" layout="responsive" width="300" height="235"></amp-dailymotion>',
+			),
+			'0_width_in_attributes'   => array(
+				'[dailymotion ' . $id . ' width=0]',
+				'<amp-dailymotion data-videoid="'. $id .'" layout="responsive" width="' . self::CONTENT_WIDTH . '" height="' . $default_height .'"></amp-dailymotion>',
+			),
+			'id_as_dailymotion_value' => array(
+				'[dailymotion=' . $id . ']',
+				'<amp-dailymotion data-videoid="'. $id .'" layout="responsive" width="' . self::CONTENT_WIDTH . '" height="' . $default_height .'"></amp-dailymotion>',
+			),
+		);
+	}
+
+	/**
+	 * Test the AMP-compatible [dailymotion] shortcode on an AMP endpoint.
+	 *
+	 * @dataProvider get_dailymotion_amp_data
+	 * @since 8.0.0
+	 *
+	 * @param string $shortcode_content The shortcode, like [dailymotion 1234].
+	 * @param string $expected The expected return value of the function.
+	 */
+	public function test_shortcodes_dailymotion_amp( $shortcode_content, $expected ) {
+		global $content_width;
```

Comment:
Could you add the following?

```suggestion
		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
			self::markTestSkipped( 'WordPress.com does not run the latest version of the AMP plugin yet.' );
			return;
		}

		global $content_width;
```